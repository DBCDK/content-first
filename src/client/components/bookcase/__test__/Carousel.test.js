import React from 'react';
import renderer from 'react-test-renderer';

import Carousel from '../Carousel.component';

jest.mock(
  '../../general/CheckmarkConnected.component',
  () => 'CheckmarkConnected'
);

const pid = '870970-basis:52038014';
const position = {x: 26, y: 46};
const description = 'lorem ipsum . . .';

const book = [
  {
    book: {
      pid: pid,
      creator: 'Nicole Boyle RÃ¸dtnes',
      title: 'test',
      cover: 'img/cover.png',
      description: description
    }
  }
];

describe('Carousel', () => {
  it('renders initial component', () => {
    const tree = renderer
      .create(
        <Carousel
          loading={true}
          description={description}
          onClick={() => {
            this.nextBook('next');
          }}
          book={book}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
