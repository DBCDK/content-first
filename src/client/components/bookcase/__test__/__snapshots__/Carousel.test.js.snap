// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders initial component 1`] = `
<div
  className="rollover-container "
>
  <div
    className="rollover"
  >
    <div
      className="col-xs-4 rollover-img"
    >
      <img
        alt=""
        className="high-contrast "
        onError={[Function]}
        src="https://metakompasset.demo.dbc.dk/api/cover/undefined"
        style={undefined}
      />
    </div>
    <div
      className="col-xs-8 text-left rollover-text"
    >
      <div
        className="col-xs-12 rollover-title"
      >
        <h1 />
      </div>
      <div
        className="col-xs-12 rollover-creator"
      >
        <h2 />
      </div>
      <div
        className="col-xs-12 rollover-description"
      >
        <p>
          lorem ipsum . . .
        </p>
      </div>
      <div
        className="col-xs-12"
      >
        <CheckmarkConnected
          book={
            Object {
              "book": Object {
                "pid": undefined,
              },
            }
          }
          origin="Fra bogreol"
        />
      </div>
    </div>
    <div
      className="clear"
    />
  </div>
</div>
`;
